spec/features/admin/invoices/show_spec.rb

```diff
@@ -14,4 +26,25 @@
     expect(page).to have_content("Customer: #{@invoice1.customer.full_name}")
     expect(page).to have_content("Created Date: #{@invoice1.date_format}")
   end
-end
\ No newline at end of file
+
+  # US 34
+  it "displays all invoice items and attributes" do
+    visit admin_invoice_path(@invoice1.id)
+
+    @invoice1.invoice_items.each do |invoice_item|
+      within("#invoice-item-#{invoice_item.id}") do
+        expect(page).to have_content("Item: #{invoice_item.item.name}")
+        expect(page).to have_content("Quantity Ordered: #{invoice_item.quantity}")
+        expect(page).to have_content("Sold Price: #{invoice_item.unit_price}")
+        expect(page).to have_content("Invoice Item Status: #{invoice_item.status}")
+      end
+    end
+  end 
+  
+  # US 35
+  it "displays the total revenue generated from the invoice" do
+    visit admin_invoice_path(@invoice1.id)
+
+    expect(page).to have_content("Total Revenue: $#{@invoice1.total_revenue}")
```

Comment:
Will have to be careful with this if other invoices are added, otherwise looks good!